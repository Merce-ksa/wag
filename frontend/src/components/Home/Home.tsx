import React from 'react'
import { Text, View, StyleSheet, StatusBar, ScrollView, TouchableOpacity } from 'react-native'

const homeStyles = StyleSheet.create({
  HomeContainer: {
    flex: 1,
    marginTop: StatusBar.currentHeight,
    paddingLeft: 20,
    paddingRight: 20,
    backgroundColor: '#FAFAFA'
  },
  title: {
    marginTop: 50,
    marginBottom: 50,
    fontSize: 24,
    textAlign: 'left',
    fontWeight: '400'
  },
  cardContent: {
    height: 60,
    marginTop: 10,
    justifyContent: 'center',
    marginBottom: 10,
    backgroundColor: '#BBCBFB',
    borderLeftColor: '#567DF4',
    borderLeftWidth: 8,
    borderTopEndRadius: 10,
    borderBottomEndRadius: 10
  },
  card: {
    fontSize: 20,
    paddingLeft: 10
  },
  floatButton: {
    width: 60,
    height: 60,
    borderRadius: 30,
    backgroundColor: '#22215B',
    position: 'absolute',
    bottom: 20,
    right: 20,
    alignItems: 'center',
    justifyContent: 'center'
  },
  textFloatButton: {
    color: '#fff',
    fontSize: 20,
    fontWeight: 'bold'
  }
})

export default function Home () {
  return (
        <View style={homeStyles.HomeContainer}>
            <Text style={homeStyles.title}>My Groups</Text>
            <ScrollView>
                <View style={homeStyles.cardContent}>
                    <Text style={homeStyles.card}>
                        My First group
                    </Text>
                </View>
                <View style={homeStyles.cardContent}>
                    <Text style={homeStyles.card}>
                        Skylab 202101
                    </Text>
                </View>
                <View style={homeStyles.cardContent}>
                    <Text style={homeStyles.card}>
                        Personal Links
                    </Text>
                </View>
                <View style={homeStyles.cardContent}>
                    <Text style={homeStyles.card}>
                        Mesenterio
                    </Text>
                </View>
                <View style={homeStyles.cardContent}>
                    <Text style={homeStyles.card}>
                        aTr√≠viate
                    </Text>
                </View>

            </ScrollView>
            <View style={homeStyles.floatButton}>
                <TouchableOpacity
                    disabled
                    onPress={() => console.log('holi')}
                >
                    <Text style={homeStyles.textFloatButton}>+</Text>
                </TouchableOpacity>
            </View>

        </View>

  )
}
