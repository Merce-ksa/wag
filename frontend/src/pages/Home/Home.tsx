/* eslint-disable react/prop-types */
import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import loadGroups from '../../redux/actions/groupsActions'
import { Text, View, ScrollView, TouchableOpacity, Image } from 'react-native'
import homeStyles from './HomeStyles'

function Home ({ groups, user, actions }) {
  console.log('info que llega a home del user:')
  console.log(user)
  console.log('grupos que llegan a la home:')
  console.log(groups)
  console.log(actions)

  useEffect(() => {
    console.log('merche')
    console.log(actions)
    actions.loadGroups()
  }, [])

  return (
        <View style={homeStyles.HomeContainer}>
          <View style={homeStyles.homeTitleContent}>
          <Image
            source={require('../../assets/images/wag-icon.png')}
            style={homeStyles.wagIcon}
          />
            <Text style={homeStyles.title}>My Groups</Text>
            <Text>Bienvenido {user.data.email}</Text>
          </View>
                {groups && groups.map((group) => (
                    <ScrollView key="groupsContainer">
                        <TouchableOpacity key={group.name} style={homeStyles.cardContent}>
                            <Text style={homeStyles.card}>
                                {group.name}
                            </Text>
                            <Text style={homeStyles.cardDate}>
                                {group.date}
                            </Text>
                        </TouchableOpacity>
                    </ScrollView>
                ))
                }

            <View style={homeStyles.floatButton}>
                <TouchableOpacity
                    disabled
                    onPress={() => console.log('holi')}
                >
                    <Text style={homeStyles.textFloatButton}>+</Text>
                </TouchableOpacity>
            </View>
        </View>
  )
}

function mapStateToProps (state) {
  console.log(state.groups)
  return {
    groups: state.groups,
    user: state.user
  }
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({ loadGroups }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
