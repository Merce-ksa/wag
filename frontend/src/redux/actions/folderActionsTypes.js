const folderActionTypes = {
  LOAD_FOLDERS: 'LOAD_FOLDERS',
  LOAD_FOLDERS_ERROR: 'LOAD_FOLDERS_ERROR',
  CREATE_FOLDER: 'CREATE_FOLDER'
}

export default folderActionTypes
