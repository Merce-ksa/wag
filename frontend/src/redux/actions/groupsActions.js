import axios from 'axios'
import groupActionsTypes from './groupActionsTypes'
import hostApp from '../../constants/host'

const host = hostApp

export function loadGroups () {
  return async (dispatch) => {
    try {
      const { data } = await axios.get(`${host}/groups`, { withCredentials: true })

      console.log(data)

      dispatch({
        type: groupActionsTypes.LOAD_GROUPS,
        groups: data
      })
    } catch {
      dispatch({
        type: groupActionsTypes.LOAD_GROUPS_ERROR
      })
    }
  }
}

export function createGroup (name, members) {
  return async (dispatch) => {
    try {
      await axios.post(`${host}/groups`, { name, members }, { withCredentials: true })

      dispatch({
        type: groupActionsTypes.CREATE_GROUP,
        groupLastUpdated: `${new Date()}`
      })
    } catch {
      dispatch({
        type: groupActionsTypes.CREATE_GROUP_ERROR,
        groupLastUpdated: null
      })
    }
  }
}
