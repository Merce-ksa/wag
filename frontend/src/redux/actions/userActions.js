import axios from 'axios'
import userActionsTypes from './userActionsTypes'
import AsyncStorage from '@react-native-community/async-storage'
// const skyHost = 'http://192.168.0.33:5000'
const homeHost = 'http://192.168.1.26:5000'

export function register (userName, email, password) {
  return async (dispatch) => {
    try {
      const user = await axios.post(`${homeHost}/auth/register`, { userName, email, password }, { withCredentials: true })
        .then(() => axios.post(`${homeHost}/auth/login`, { email, password }, { withCredentials: true }))
        .then(() => axios.get(`${homeHost}/user/me`, { withCredentials: true }))

      AsyncStorage.setItem('user', JSON.stringify(user.data))

      dispatch({
        type: userActionsTypes.REGISTERED,
        data: user.data
      })
    } catch {
      console.log('registro fallido')

      dispatch({
        type: userActionsTypes.REGISTERED_ERROR
      })
    }
  }
}

export function login (email, password) {
  return async (dispatch) => {
    try {
      const user = await axios.post(`${homeHost}/auth/login`, { email, password }, { withCredentials: true })
        .then(() => axios.get(`${homeHost}/user/me`, { withCredentials: true }))

      AsyncStorage.setItem('user', JSON.stringify(user.data))

      dispatch({
        type: userActionsTypes.LOGIN,
        data: user.data
      })
    } catch {
      dispatch({
        type: userActionsTypes.LOGIN_ERROR
      })
    }
  }
}

export function logout () {
  return async (dispatch) => {
    await axios.post(`${homeHost}/auth/logout`, { withCredentials: true })
    AsyncStorage.removeItem('user')

    dispatch({
      type: userActionsTypes.LOGOUT
    })
  }
}

export function loadUserFromStorage () {
  return async (dispatch) => {
    const userFromStorage = await AsyncStorage.getItem('user')
    const user = JSON.parse(userFromStorage)

    dispatch({
      type: userActionsTypes.LOAD_USER,
      user
    })
  }
}
