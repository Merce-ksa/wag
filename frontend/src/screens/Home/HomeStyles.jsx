import {
  StyleSheet
} from 'react-native'

export default StyleSheet.create({
  groupsContainer: {
    backgroundColor: '#fff',
    padding: 10,
    borderRadius: 20
  },
  titleGroups: {
    fontFamily: 'interSemiBold',
    fontSize: 18
  },
  subTitleGroups: {
    marginTop: 5,
    fontFamily: 'interRegular',
    fontSize: 16
  },
  cardContent: {
    height: 60,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',

    marginTop: 15,
    marginBottom: 15,
    marginLeft: 5,
    marginRight: 5,
    padding: 10,

    borderTopEndRadius: 10,
    borderBottomEndRadius: 10,

    borderRadius: 5,
    backgroundColor: '#fff',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4
    },
    shadowOpacity: 0.30,
    shadowRadius: 4.65,

    elevation: 6
  },
  infoGroupRight: {
    marginRight: 20
  },
  card: {
    fontFamily: 'interMedium',
    paddingLeft: 10,
    fontSize: 20,
    color: '#22215B'
  },
  cardDate: {
    paddingLeft: 10,
    fontSize: 14,
    fontWeight: 'bold',
    color: '#7B7F9E'
  },
  arrowGroup: {
    fontFamily: 'interRegular',
    fontSize: 18
  },
  newGroupContainer: {
    height: 150,
    marginTop: 30,
    borderRadius: 20,
    backgroundColor: '#fff'
  },
  newGroupButtonContainer: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  newGroupButton: {
    width: 150,
    height: 38,
    margin: 15,
    justifyContent: 'center',
    padding: 10,
    borderRadius: 5,
    backgroundColor: '#567DF4'
  },
  textGroupButton: {
    color: '#fff',
    textAlign: 'center',
    fontSize: 16
  }
})
