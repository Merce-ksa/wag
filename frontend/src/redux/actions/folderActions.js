import axios from 'axios'
import folderActionsTypes from './folderActionsTypes'
import hostApp from '../../constants/host'

const host = hostApp

export function loadFolders (groupId) {
  return async (dispatch) => {
    try {
      const allFoldersGroup = await axios.get(`${host}/folder/${groupId}`, { withCredentials: true })
      console.log(allFoldersGroup)

      dispatch({
        type: folderActionsTypes.LOAD_FOLDERS,
        folders: allFoldersGroup.data
      })
    } catch {
      dispatch({
        type: folderActionsTypes.LOAD_FOLDERS_ERROR,
        folders: null
      })
    }
  }
}

export function createFolder (name, colorSelected, groupId) {
  console.log(colorSelected.color)
  const color = colorSelected.color
  return async (dispatch) => {
    try {
      await axios.post(`${host}/folder/`, { name, color, groupId }, { withCredentials: true })

      dispatch({
        type: folderActionsTypes.CREATE_FOLDER,
        folderLastUpdated: `${new Date()}`
      })
    } catch {
      dispatch({
        type: folderActionsTypes.CREATE_FOLDER_ERROR,
        folderLastUpdated: null
      })
    }
  }
}
