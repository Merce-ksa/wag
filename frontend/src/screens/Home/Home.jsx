import React, { useEffect } from 'react'
import {
  Text,
  View,
  ScrollView,
  TouchableOpacity,
  Image
} from 'react-native'
import { useIsFocused } from '@react-navigation/native'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { loadUserFromStorage, logout } from '../../redux/actions/userActions'
import { loadGroups } from '../../redux/actions/groupsActions'
import homeStyles from './HomeStyles'
import bodyStyles from '../../assets/styles/bodyStyles'

function Home ({ groups, user, actions, groupLastUpdated, navigation }) {
  const isFocused = useIsFocused()
  console.log('Entra al componente')

  useEffect(() => {
    actions.loadUserFromStorage()
  }, [])

  useEffect(() => {
    actions.loadGroups()
  }, [isFocused])

  useEffect(() => {
    actions.loadGroups()
  }, [groupLastUpdated])

  if (!user.email) {
    navigation.navigate('Auth')
  }

  return (
    <View style={bodyStyles.container}>
      <View style={bodyStyles.titleContent}>
      <Image
        key="icon-groups"
        source={require('../../assets/images/groups.png')}
        style={bodyStyles.sectionIcon}
      />
        <Text style={bodyStyles.title}>Hi {user.userName}!</Text>

        <TouchableOpacity
          testID="profile-navigate"
          onPress={() =>
            navigation.navigate('Profile')
          }
        >
        </TouchableOpacity>

      </View>
      <ScrollView key="groupsContainer">
        <View style={homeStyles.groupsContainer}>
          <Text style={homeStyles.titleGroups}>My groups</Text>
        {groups && groups.map((group) => (
          <TouchableOpacity
            key={group.name}
            style={homeStyles.cardContent}
            onPress={() => navigation.navigate('FolderList', {
              groupId: group.groupId,
              groupName: group.name
            })}
            testID="group-name"
          >
            <View style={homeStyles.infoGroupRight}>
              <Text style={homeStyles.card} testID="group-name">
                    {group.name}
                </Text>
                <Text style={homeStyles.cardDate}>
                    {group.date}
                </Text>
            </View>
            <View style={homeStyles.infoGroupRight}>
              <Text style={homeStyles.arrowGroup}>â†’</Text>
            </View>

          </TouchableOpacity>
        ))}
        </View>
        <View style={[homeStyles.groupsContainer, homeStyles.newGroupContainer]}>
            <Text style={homeStyles.titleGroups}>Create your own group</Text>
            <Text style={homeStyles.subTitleGroups}>
                Share a world of resources with your friends.
            </Text>
            <View style={homeStyles.newGroupButtonContainer}>
              <TouchableOpacity
                style={homeStyles.newGroupButton}
                onPress={() => navigation.navigate('NewGroup')}
                testID="group-navigate"
              >
                <Text style={homeStyles.textGroupButton}>Create group</Text>
            </TouchableOpacity>
            </View>
        </View>
      </ScrollView>
  </View>
  )
}

Home.propTypes = {
  actions: PropTypes.shape({
    loadGroups: PropTypes.func.isRequired,
    loadUserFromStorage: PropTypes.func.isRequired
  }).isRequired,

  user: PropTypes.shape({
    userName: PropTypes.string,
    email: PropTypes.string
  }),

  groups: PropTypes.arrayOf(PropTypes.shape({
    groupId: PropTypes.string,
    members: PropTypes.arrayOf(
      PropTypes.string
    ),
    name: PropTypes.string,
    date: PropTypes.string
  })),

  groupLastUpdated: PropTypes.string,

  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired
}

function mapStateToProps (state) {
  return {
    groups: state.groups,
    user: state.user,
    groupLastUpdated: state.groupLastUpdated
  }
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({ loadGroups, logout, loadUserFromStorage }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
