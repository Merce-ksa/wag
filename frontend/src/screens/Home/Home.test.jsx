import React from 'react'
import { Provider } from 'react-redux'
import * as groupsActions from '../../redux/actions/groupsActions'
import * as userActions from '../../redux/actions/userActions'
import { render, waitFor, fireEvent } from '@testing-library/react-native'
import '@testing-library/jest-dom'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Home from './Home'

const mockStore = configureStore([thunk])

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useIsFocused: () => false
}))

jest.mock('@react-native-community/async-storage', () => ({
  getItem: jest.fn(() => Promise.resolve()),
  setItem: jest.fn(() => Promise.resolve())
}))

describe('Given a Home function', () => {
  let store
  let navigation
  beforeEach(() => {
    jest.spyOn(groupsActions, 'loadGroups').mockReturnValueOnce({ type: '' })
    jest.spyOn(userActions, 'loadUserFromStorage').mockReturnValueOnce({ type: '' })
    store = mockStore({
      groups: [
        {
          grouId: 'firstNumber',
          name: 'first Group Name',
          date: 'mockDate'
        },
        {
          grouId: 'secondNumber',
          name: 'second Group Name',
          date: 'mockDate'
        }
      ],
      user: {
        userName: 'mockName'
      },
      groupLastUpdated: 'mockNewDate'
    })

    navigation = {
      navigate: jest.fn()
    }
  })

  describe('When is rendered', () => {
    test('Should list user groups', async () => {
      const { getAllByTestId } = await waitFor(() => render(<Provider store={store}><Home navigation = { navigation } /></Provider>))

      expect(getAllByTestId('group-name').length).toBe(2)
      expect(getAllByTestId('group-name')[0].props.children).toBe('first Group Name')
    })
  })

  describe('When the group button is pressed', () => {
    test('Then navigation should invoked', async () => {
      const { getAllByTestId } = await waitFor(() => render(<Provider store={store}><Home navigation = { navigation } /></Provider>))

      fireEvent.press(getAllByTestId('group-navigate')[0])
      console.log(getAllByTestId('group-navigate')[0].props)
      expect(getAllByTestId('group-navigate')[0].props).toBeDefined()
    })
  })

  describe('When the profile button is pressed', () => {
    test('Then navigation should invoked', async () => {
      const { getByTestId } = await waitFor(() => render(<Provider store={store}><Home navigation = { navigation } /></Provider>))

      fireEvent.press(getByTestId('profile-navigate'))

      expect(getByTestId('profile-navigate').props).toBeDefined()
    })
  })

  describe('When the New Group button is pressed', () => {
    test('Then navigation should invoked', async () => {
      const { getByTestId } = await waitFor(() => render(<Provider store={store}><Home navigation = { navigation } /></Provider>))

      fireEvent.press(getByTestId('group-navigate'))

      expect(getByTestId('group-navigate').props).toBeDefined()
    })
  })
})
