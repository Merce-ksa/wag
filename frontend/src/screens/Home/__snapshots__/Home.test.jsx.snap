// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Home function When is rendered Then the SnapShot should match the rendered tree 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "marginTop": null,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "marginRight": 10,
          "resizeMode": "contain",
          "width": 50,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "barlowMedium",
          "fontSize": 30,
          "fontWeight": "400",
          "marginBottom": 50,
          "marginTop": 50,
          "textAlign": "left",
        }
      }
    >
      Hi 
      mockName
      !
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    />
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "interSemiBold",
              "fontSize": 18,
            }
          }
        >
          My groups
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomEndRadius": 10,
              "borderRadius": 5,
              "borderTopEndRadius": 10,
              "elevation": 6,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
              "marginBottom": 15,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 15,
              "opacity": 1,
              "padding": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#22215B",
                  "fontFamily": "interMedium",
                  "fontSize": 20,
                  "paddingLeft": 10,
                }
              }
            >
              mockGroupName
            </Text>
            <Text
              style={
                Object {
                  "color": "#7B7F9E",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "paddingLeft": 10,
                }
              }
            >
              mockDate
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "interRegular",
                  "fontSize": 18,
                }
              }
            >
              →
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomEndRadius": 10,
              "borderRadius": 5,
              "borderTopEndRadius": 10,
              "elevation": 6,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
              "marginBottom": 15,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 15,
              "opacity": 1,
              "padding": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#22215B",
                  "fontFamily": "interMedium",
                  "fontSize": 20,
                  "paddingLeft": 10,
                }
              }
            >
              mockGroupName2
            </Text>
            <Text
              style={
                Object {
                  "color": "#7B7F9E",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "paddingLeft": 10,
                }
              }
            >
              mockDate
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "interRegular",
                  "fontSize": 18,
                }
              }
            >
              →
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 20,
              "padding": 10,
            },
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 20,
              "height": 150,
              "marginTop": 30,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "interSemiBold",
              "fontSize": 18,
            }
          }
        >
          Create your own group
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "interRegular",
              "fontSize": 16,
              "marginTop": 5,
            }
          }
        >
          Share a world of resources with your friends.
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#567DF4",
                "borderRadius": 5,
                "height": 38,
                "justifyContent": "center",
                "margin": 15,
                "opacity": 1,
                "padding": 10,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              Create group
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
